src/vm/methodtable.h

```diff
@@ -2195,33 +2203,59 @@ class MethodTable
         LIMITED_METHOD_DAC_CONTRACT;
 
         PRECONDITION(IsParentMethodTablePointerValid());
-
-        TADDR pMT = m_pParentMethodTable;
-#ifdef FEATURE_PREJIT
-        if (GetFlag(enum_flag_HasIndirectParent))
-            pMT = *PTR_TADDR(m_pParentMethodTable + offsetof(MethodTable, m_pParentMethodTable));
-#endif
-        return PTR_MethodTable(pMT);
+        return ReadPointerMaybeNull(this, &MethodTable::m_pParentMethodTable, GetFlagHasIndirectParent());
     }
 
     inline static PTR_VOID GetParentMethodTableOrIndirection(PTR_VOID pMT)
     {
         WRAPPER_NO_CONTRACT;
-        return PTR_VOID(*PTR_TADDR(dac_cast<TADDR>(pMT) + offsetof(MethodTable, m_pParentMethodTable)));
+        PTR_MethodTable pMethodTable = dac_cast<PTR_MethodTable>(pMT);
+        TADDR base = dac_cast<TADDR>(pMethodTable) + offsetof(MethodTable, m_pParentMethodTable);
+        return pMethodTable->m_pParentMethodTable.GetValueOrIndirectionMaybeNull(base);
```

Comment:
I do not think this will work correctly for `DACCESS_BUILD` build.

`pMethodTable` does not necessarily point to valid memory when it is the indirection. `pMethodTable->` in `DACCESS_BUILD` will cause the MethodTable to be materialize on the debugger side. The operation will fail when `pMethodTable` does not point to valid `sizeof(MethodTable)` worth of valid memory.

Reply:
I have replaced this with `#ifdef`, with modified version for Linux ARM, and original version for other platforms. What do you think?

Reply:
Yes, this looks ok. Thanks.